src/RestControllers/AuthorizationController.php

```diff
@@ -62,9 +64,13 @@ class AuthorizationController
     public function __construct($providerForm = true)
     {
         global $gbl;
+
         $this->siteId = $gbl::$SITE;
-        $this->serverBaseUrl = $gbl::$REST_BASE_URL;
-        $this->authBaseUrl = $gbl::$REST_FULLAUTH_URL;
+
+        $this->authBaseUrl = $GLOBALS['webroot'] . '/oauth2/' . $_SESSION['site_id'];
+        // collect full url by using 'site_addr_oath' global
+        $this->authBaseFullUrl = $GLOBALS['site_addr_oath'] . $this->authBaseUrl;
```

Comment:
Again, I worked hard to keep globals out of this. No need for them and besides these should be set in class properties and carried around i.e short lived.